runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MusicClue.java

```diff
@@ -0,0 +1,104 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.cluescrolls.clues;
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import lombok.Getter;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.plugins.cluescrolls.ClueScrollPlugin;
+import net.runelite.client.ui.overlay.OverlayUtil;
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+import static net.runelite.client.plugins.cluescrolls.ClueScrollOverlay.TITLED_CONTENT_COLOR;
+import static net.runelite.client.plugins.cluescrolls.ClueScrollWorldOverlay.IMAGE_Z_OFFSET;
+
+@Getter
+public class MusicClue extends ClueScroll implements NpcClueScroll
+{
+	private static WorldPoint LOCATION = new WorldPoint(2990, 3384, 0);
+	private static String NPC = "Cecilia";
+	private static Pattern SONG_PATTERN = Pattern.compile("<col=ffffff>([A-Za-z ]+)</col>");
+
+	private String song;
+
+	private MusicClue(String song)
+	{
+		this.song = song;
+	}
+
+	@Override
+	public void makeOverlayHint(PanelComponent panelComponent, ClueScrollPlugin plugin)
+	{
+		panelComponent.getChildren().add(TitleComponent.builder().text("Music Clue").build());
+		panelComponent.getChildren().add(LineComponent.builder().left("NPC:").build());
+		panelComponent.getChildren().add(LineComponent.builder()
+			.left(NPC)
+			.leftColor(TITLED_CONTENT_COLOR)
+			.build());
+
+		panelComponent.getChildren().add(LineComponent.builder().left("Area:").build());
+		panelComponent.getChildren().add(LineComponent.builder()
+			.left("Falador Park")
+			.leftColor(TITLED_CONTENT_COLOR)
+			.build());
+
+		panelComponent.getChildren().add(LineComponent.builder().left("Song:").build());
+		panelComponent.getChildren().add(LineComponent.builder()
+			.left(song)
+			.leftColor(TITLED_CONTENT_COLOR)
+			.build());
+	}
+
+	@Override
+	public void makeWorldOverlayHint(Graphics2D graphics, ClueScrollPlugin plugin)
+	{
+		if (!LOCATION.isInScene(plugin.getClient()))
+		{
+			return;
+		}
+
+		for (net.runelite.api.NPC npc : plugin.getNpcsToMark())
+		{
+			OverlayUtil.renderActorOverlayImage(graphics, npc, plugin.getClueScrollImage(), Color.ORANGE, IMAGE_Z_OFFSET);
+		}
+	}
+
+	public static MusicClue forText(String text)
+	{
+		final Matcher m = SONG_PATTERN.matcher(text);
+		m.find();
+		final String song = m.group(1);
+		return new MusicClue(song);
+	}
+
+	public String[] getNpcs()

```

Comment:
?

Reply:
This is so the Clue Plugin will highlight Cecilia